### PR TITLE
[NOT MERGE] Verify UI testing in CI server

### PR DESCRIPTION
**Pull request by wezhang**

### CODE REVIEW

**Comment by vscjavaci** [comment]
Can one of the admins verify this patch?

**Comment by vscjavaci** [comment]
Can one of the admins verify this patch?

**Comment by wezhang** [comment]
```
[ERROR]     Unresolveable build extension: Plugin org.eclipse.tycho:tycho-maven-plugin:0.25.0 or one of its dependencies could not be resolved: The following artifacts could not be resolved: org.eclipse.tycho:tycho-maven-plugin:jar:0.25.0, org.codehaus.plexus:plexus-utils:jar:1.1: Cannot access central (https://repo.maven.apache.org/maven2) in offline mode and the artifact org.eclipse.tycho:tycho-maven-plugin:jar:0.25.0 has not been downloaded from it before. -> [Help 2]
```

**Comment by wezhang** [comment]
Only has tycho-maven-plugin 1.0.0 in maven repo

**Comment by vscjavaci** [comment]
Can one of the admins verify this patch?

**Comment by vscjavaci** [comment]
Can one of the admins verify this patch?

File: Utils/hdinsight-node-common/pom.xml [hunk 1]
```diff
@@ -88,6 +88,18 @@
                     <target>1.8</target>
                 </configuration>
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
```
Comment (voidfunction) [inline Utils/hdinsight-node-common/pom.xml]: it's offline build now. build failed can be expected if new references added.

Comment (wezhang) [inline Utils/hdinsight-node-common/pom.xml]: it reported from `[ERROR]   The project com.microsoft.azuretools.wap4ej.build:parent:3.0.4-SNAPSHOT (/home/travis/build/Microsoft/azure-tools-for-java/PluginsAndFeatures/azure-toolkit-for-eclipse/pom.xml) has 1 error`

Comment (wezhang) [inline Utils/hdinsight-node-common/pom.xml]: #595 also hit.

**Comment by wezhang** [comment]
Looks like Graphics interface is missing;

```
    And open spark submission dialog                                                                               # SparkSubmissionExDialogScenario.openSparkSubmissionDialog()
      java.awt.HeadlessException: 
      No X11 DISPLAY variable was set, but this program performed an operation which requires it.
      	at java.awt.GraphicsEnvironment.checkHeadless(GraphicsEnvironment.java:207)
      	at java.awt.Window.<init>(Window.java:536)
      	at java.awt.Frame.<init>(Frame.java:420)
      	at java.awt.Frame.<init>(Frame.java:385)
      	at javax.swing.SwingUtilities$SharedOwnerFrame.<init>(SwingUtilities.java:1758)
      	at javax.swing.SwingUtilities.getSharedOwnerFrame(SwingUtilities.java:1833)
      	at javax.swing.JDialog.<init>(JDialog.java:272)
      	at javax.swing.JDialog.<init>(JDialog.java:206)
      	at javax.swing.JDialog.<init>(JDialog.java:154)
      	at com.microsoft.azure.hdinsight.spark.ui.SparkSubmissionExDialog.<init>(SparkSubmissionExDialog.java:68)
      	at com.microsoft.azure.hdinsight.spark.ui.SparkSubmissionExDialogScenario.openSparkSubmissionDialog(SparkSubmissionExDialogScenario.java:109)
      	at ✽.And open spark submission dialog(com/microsoft/azure/hdinsight/spark/ui/SparkSubmissionExDialogScenario.feature:8)
```

**Comment by wezhang** [comment]
2016 passed, 2017 failed.

```
Failed scenarios:
com/microsoft/azure/hdinsight/spark/ui/SparkSubmissionExDialogScenario.feature:3 # Scenario: Test the submission parameters with default settings
2 Scenarios (1 failed, 1 passed)
9 Steps (1 failed, 5 skipped, 3 passed)
0m7.251s
java.lang.NullPointerException
	at com.microsoft.azure.hdinsight.spark.ui.SparkSubmissionContentPanel.initializeModel(SparkSubmissionContentPanel.java:219)
	at com.microsoft.azure.hdinsight.spark.ui.SparkSubmissionContentPanel.initialize(SparkSubmissionContentPanel.java:177)
	at com.microsoft.azure.hdinsight.spark.ui.SparkSubmissionContentPanel.<init>(SparkSubmissionContentPanel.java:77)
	at com.microsoft.azure.hdinsight.spark.ui.SparkSubmissionExDialog.initializeComponents(SparkSubmissionExDialog.java:93)
	at com.microsoft.azure.hdinsight.spark.ui.SparkSubmissionExDialog.<init>(SparkSubmissionExDialog.java:74)
	at com.microsoft.azure.hdinsight.spark.ui.SparkSubmissionExDialogScenario.openSparkSubmissionDialog(SparkSubmissionExDialogScenario.java:109)
	at ✽.And open spark submission dialog(com/microsoft/azure/hdinsight/spark/ui/SparkSubmissionExDialogScenario.feature:8)
```

**Comment by vscjavaci** [comment]
Can one of the admins verify this patch?

**Comment by wezhang** [comment]
AzureActionsComponent() async initilization may be slow. So, getIdeHelper() returned NULL.

**Comment by wezhang** [comment]
slf4j is different between 2016 and 2017 testing:

| 2016 | 2017|
|-------|------|
| `SLF4J: Actual binding is of type [org.slf4j.impl.Log4jLoggerFactory]` <br /> `Logger path:/home/travis/AzureToolsForIntelliJ/corelibs.log` <br /> `Jun 07, 2017 1:37:15 AM java.util.logging.LogManager$RootLogger log` <br /> `INFO: === Log session started ===` <br /> `AI: INFO 07-06-2017 01:37, 25: Configuration file has been successfully found as resource` | `SLF4J: Actual binding is of type [org.slf4j.impl.Log4jLoggerFactory]` |

**Comment by wezhang** [comment]
Now, 2016 failed.

**Comment by wezhang** [comment]
Reproed in local, get the following errors between two builds:

```
2017-06-07 13:26:59,772 [      0]   WARN - .ide.plugins.PluginManagerCore - Can't load PluginDescriptor[name='Azure Toolkit for IntelliJ', classpath='D:\src\github\Microsoft\my\azure-tools-for-java\PluginsAndFeatures\azure-toolkit-for-intellij\build\resources\main']: until build 163.SNAPSHOT does not match IC-171.3780.107 
2017-06-07 13:26:59,773 [      1]   INFO - llij.ide.plugins.PluginManager - Plugin 'Azure Toolkit for IntelliJ' can't be loaded because: Plugin since-build or until-build don't match this product's build number 
```

**Comment by wezhang** [comment]
Can be fixed by this:

```
diff --git a/BuildDevint b/BuildDevint
index c2cb1e6..5b25a90 100644
--- a/BuildDevint
+++ b/BuildDevint
@@ -46,6 +46,8 @@ fi
 ./gradlew --project-dir ./PluginsAndFeatures/azure-toolkit-for-intellij -s
 cp ./PluginsAndFeatures/azure-toolkit-for-intellij/build/distributions/azure-toolkit-for-intellij.zip ./$ARTIFACTS_DIR/azure-toolkit-for-intellij-2017.zip

+./gradlew clean --project-dir ./PluginsAndFeatures/azure-toolkit-for-intellij -s
+
 # Build intellij 2016 plugin
 if [ $INTELLIJ_VERSION == "true" ] ; then
     ./tools/IntellijVersionHelper 2016.3
```

**Comment by vscjavaci** [comment]
Can one of the admins verify this patch?

**Comment by Eskibear** [comment]
shall we close this PR?  @wezhang

**Pull request closed without merging**